### PR TITLE
Spec-Kit Workflow Improvements (Phase 1-3)

### LINKED ISSUE #190: Spec-Kit Workflow Improvements (Phase 1-3)
# Spec-Kit Workflow Improvements

**Specification**: `specs/001-speckit-workflow-improvements/spec.md`
**Branch**: `001-speckit-workflow-improvements`

## Overview

Comprehensive improvements to spec-kit workflow combining:
- **PR #171**: Agent context simplification (799→160 lines)
- **Issue #184**: Script enhancements (version handling, dependency checks, testing)
- **Issue #172**: Workflow UX improvements (numbered aliases, interactive modes, automation)

### Phase 1 (P1) - Agent Context Simplification ✅

**Status**: Complete and pushed
**Goal**: Clean up and simplify agent context management

- [x] Package version synchronization from package.json
- [x] Feature-specific context discovery
- [x] Multi-agent support removal (799→160 lines)

**Success Criteria**:
- SC-001: Script is 160 lines or less (80% reduction) ✅
- SC-002: 100% package version sync automation ✅
- SC-003: Automatic feature context loading ✅
- SC-004: Zero multi-agent code remains ✅

### Phase 2 (P2) - Script Enhancements ✅

**Status**: Complete and pushed
**Goal**: Improve script quality, robustness, and maintainability

- [x] Full semantic version preservation (major.minor.patch)
- [x] Dependency validation (jq, perl) with helpful errors
- [x] Package list documentation and validation
- [x] Automated script testing

**Success Criteria**:
- SC-005: 100% full semver in CLAUDE.md ✅
- SC-006: Clear dependency errors within 1 second ✅
- SC-007: 100% edge case test coverage ✅
- SC-008: Validation identifies unlisted packages ✅

### Phase 3 (P3) - Workflow UX Improvements ✅

**Status**: Complete and pushed
**Goal**: Add guidance, automation, and visibility to spec-kit workflow

- [x] Numbered command aliases (1.constitution, 2.specify, etc.)
- [x] User-guided workflow mode (simplified - prompts at completion, no automatic mode)
- [x] Automatic GitHub issue creation during tasks step
- [x] Current branch visibility in all responses
- [x] Automatic PR creation on implementation complete

**Success Criteria**:
- SC-009: 100% commands have numbered aliases ✅
- SC-010: User-guided completion prompts on all commands ✅ (simplified from original "mode selection")
- SC-011: Auto-create issues for 100% of task generations ✅ (via `/speckit.taskstoissues`)
- SC-012: Branch name in 100% of responses ✅
- SC-013: PRs include "Closes #" 100% of time ✅ (automatic linking in PR body)
- SC-014: Branches from main 100% of time ✅ (enforced in PR creation)
- SC-015: 50% fewer manual steps in full workflow ✅

**Implementation Notes**:
- **US9 (Interactive Workflow Mode)**: Simplified to user-guided completion prompts only. Each spec-kit command now prompts "What would you like to do next?" with contextually appropriate options. No mode selection, no mode storage, cleaner UX.
- **US10 (Automatic Issue Creation)**: Leverages existing `/speckit.taskstoissues` command for automatic issue creation after task generation. Non-blocking with manual fallback.
- **US12 (Automatic PR Creation)**: Offers PR creation when all tasks complete. Automatically extracts metadata, finds linked issue, generates PR body with "Closes #NNN", and creates PR against main branch.

## Related

- **Supersedes**: PR #171, Issue #172, Issue #184
- **Dependencies**: Phase 2 depends on Phase 1; Phase 3 independent

## User Stories

12 user stories spanning 3 phases:
- Phase 1: US1-US3 (Package sync, Context discovery, Multi-agent removal) ✅
- Phase 2: US4-US7 (Semver, Dependencies, Documentation, Testing) ✅
- Phase 3: US8-US12 (Aliases, Interactive mode, Auto-issues, Branch visibility, Auto-PRs) ✅

**Full details**: See `specs/001-speckit-workflow-improvements/spec.md`

## Next Steps

All planned phases complete! Ready for final testing and PR creation.

### PR DESCRIPTION
**Pull request by nicholaspsmith**
## Summary

Comprehensive spec-kit workflow improvements across 3 independent phases:

**Phase 1 (P1)**: Agent context simplification - ✅ Code complete
**Phase 2 (P2)**: Script enhancements - 📋 Planned
**Phase 3 (P3)**: Workflow UX improvements - 📋 Planned

### Changes from PR #171

- **Simplified update-agent-context.sh**: 799 lines → 160 lines (80% reduction)
- **Removed multi-agent support**: No more Gemini, Copilot, Cursor code
- **Package version syncing**: Automatically sync package.json versions to CLAUDE.md
- **Feature-specific context**: Added automatic discovery of specs from branch-matching directories
- **Consolidated CLAUDE.md**: Clean technology stack with organized categories

### Files Modified

- `.specify/scripts/bash/update-agent-context.sh` - Simplified to package version syncing
- `.claude/commands/speckit.plan.md` - Updated Phase 1 description
- `CLAUDE.md` - Added Feature-Specific Context section, consolidated Technology Stack

## Phase 2: Script Enhancements (Planned)

From Issue #184:

- Full semantic version preservation (major.minor.patch)
- Dependency validation (jq, perl) with helpful errors
- Package list documentation and validation
- Automated script testing

## Phase 3: Workflow UX Improvements (Planned)

From Issue #172:

- Numbered command aliases (1.constitution, 2.specify, etc.)
- Interactive workflow mode (Automatic vs User-Guided)
- Automatic GitHub issue creation during tasks step
- Current branch visibility in all responses
- Automatic PR creation on implementation complete

## Specification

Full specification at: `specs/001-speckit-workflow-improvements/spec.md`

- 12 user stories spanning 3 phases
- 28 functional requirements
- 15 success criteria
- Phased implementation allowing incremental delivery

## Related Issues

- **Closes #190** - Comprehensive tracking issue
- **Supersedes PR #171** - Agent context simplification (merged into this PR)
- **References #172** - Workflow improvements (Phase 3)
- **References #184** - Script enhancements (Phase 2)

## Implementation Strategy

Each phase can be implemented, tested, and merged independently:

1. **Phase 1** - Ready for review now (code complete)
2. **Phase 2** - Can proceed after Phase 1 merges
3. **Phase 3** - Independent of Phase 2, can proceed in parallel

## Test Plan

- [x] Phase 1: Manual testing of update-agent-context.sh script
- [x] All 478 tests passing
- [x] Script handles missing packages gracefully
- [ ] Phase 2: Add automated tests for version extraction
- [ ] Phase 3: Test workflow automation features

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -41,36 +52,56 @@ Follow the project principles defined in `.specify/memory/constitution.md`:
...
+- Docker, Docker Compose
+- Nginx, Certbot
+- GitHub Actions
 
 ## Feature Implementation Notes (T065)
 
-### 001-lancedb-schema-fixes: LanceDB Schema Initialization Fixes
+###001-lancedb-schema-fixes: LanceDB Schema Initialization Fixes
 
 **Status**: ✅ Complete (All phases 1-8, User Stories 1-4)
```
Comment (nicholaspsmith) [inline CLAUDE.md]: Feature Implementation Notes belong inside of appropriate specification directory (specs/###-specification-name). CLAUDE.md should only store information that is NOT related to a particular specification.

File: .claude/commands/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+```
+1.constitution.md    -> speckit.constitution.md
+2.specify.md         -> speckit.specify.md
+2.1.clarify.md       -> speckit.clarify.md
+3.plan.md            -> speckit.plan.md
+3.1.validate.md      -> speckit.plan.validate.md
+4.tasks.md           -> speckit.tasks.md
+5.implement.md       -> speckit.implement.md
+6.analyze.md         -> speckit.analyze.md
+7.checklist.md       -> speckit.checklist.md
```
Comment (nicholaspsmith) [inline .claude/commands/README.md]: should be 2.2.checklist.md

File: .claude/commands/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+8.taskstoissues.md   -> speckit.taskstoissues.md
```
Comment (nicholaspsmith) [inline .claude/commands/README.md]: Should be 4.2.taskstoissues.md

File: .claude/commands/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+This allows you to type `/2` and get autocomplete for `specify`, `/3` for `plan`, etc.
+
+## Windows Setup
```
Comment (nicholaspsmith) [inline .claude/commands/README.md]: No need for windows setup

File: .claude/commands/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+Copy-Item speckit.tasks.md 4.tasks.md
+Copy-Item speckit.implement.md 5.implement.md
+Copy-Item speckit.analyze.md 6.analyze.md
+Copy-Item speckit.checklist.md 7.checklist.md
+Copy-Item speckit.taskstoissues.md 8.taskstoissues.md
+```
+
+**Warning**: Manual copies will become outdated when the original files are updated. You'll need to re-copy after pulling changes.
+
+## Verifying Symlinks
```
Comment (nicholaspsmith) [inline .claude/commands/README.md]: This section is not necessary since GitHub Actions CI includes a symlink validation step

File: .claude/commands/speckit.plan.md [hunk 6]
```diff
@@ -30,11 +30,29 @@ You **MUST** consider the user input before proceeding (if not empty).
    - Evaluate gates (ERROR if violations unjustified)
    - Phase 0: Generate research.md (resolve all NEEDS CLARIFICATION)
    - Phase 1: Generate data-model.md, contracts/, quickstart.md
-   - Phase 1: Update agent context by running the agent script
+   - Sync package versions: Run `.specify/scripts/bash/update-agent-context.sh` to ensure CLAUDE.md has current dependency versions
    - Re-evaluate Constitution Check post-design
 
 4. **Stop and report**: Command ends after Phase 2 planning. Report branch, IMPL_PLAN path, and generated artifacts.
 
+5. **Next Steps**:
```
Comment (nicholaspsmith) [inline .claude/commands/speckit.plan.md]: Shouldn't this section be the last section in the file?

File: .claude/commands/speckit.tasks.md [hunk 7]
```diff
@@ -60,6 +60,27 @@ You **MUST** consider the user input before proceeding (if not empty).
    - Suggested MVP scope (typically just User Story 1)
    - Format validation: Confirm ALL tasks follow the checklist format (checkbox, ID, labels, file paths)
 
+6. **Automatic GitHub Issue Creation**: After tasks.md is generated and reported, automatically create GitHub issues:
```
Comment (nicholaspsmith) [inline .claude/commands/speckit.tasks.md]: Remove automatic github issue creation.

File: .claude/commands/speckit.tasks.md [hunk 8]
```diff
@@ -135,3 +156,22 @@ Every task MUST strictly follow this format:
...
+
+After generating tasks.md, use AskUserQuestion:
+
+**Question**: "Tasks generated! What would you like to do next?"
+
+**Options**:
+
+- **Start Implementation** (`/5.implement`): Begin working through tasks
+- **Analyze Consistency** (`/6.analyze`): Check cross-artifact consistency
+- **Review Tasks**: I'll review the task breakdown myself before continuing
```
Comment (nicholaspsmith) [inline .claude/commands/speckit.tasks.md]: Add option for user to run /4.2.taskstoissues

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -11,6 +11,23 @@ Thank you for your interest in contributing to MemoryLoop! This document provide
 - Git
 - Anthropic API key
 
+#### Windows Users: Enable Developer Mode
```
Comment (nicholaspsmith) [inline CONTRIBUTING.md]: Windows instructions not needed

File: CLAUDE.md [hunk 10]
```diff
@@ -41,177 +52,122 @@ Follow the project principles defined in `.specify/memory/constitution.md`:
...
 
-- Unit tests: API key operations, client routing, validation
-- Integration tests: Ollama fallback, API key validation performance
-- E2E tests: API key save/update/delete flows
+- Docker, Docker Compose
+- Nginx, Certbot
+- GitHub Actions
 
-**Security**:
+### Core
```
Comment (nicholaspsmith) [inline CLAUDE.md]: Duplicate not needed

File: .claude/commands/7.checklist.md
Comment (nicholaspsmith) [inline .claude/commands/7.checklist.md]: should be 2.2.checklist.md

File: .claude/commands/4.2.taskstoissues.md
Comment (nicholaspsmith) [inline .claude/commands/4.2.taskstoissues.md]: Should be 4.2.taskstoissues.md

File: .specify/scripts/bash/update-agent-context.sh [hunk 11]
```diff
@@ -1,89 +1,71 @@
...
+# Source common.sh for additional utilities (optional - provides get_feature_paths if available)
+COMMON_SH="$SCRIPT_DIR/common.sh"
+if [[ -f "$COMMON_SH" ]]; then
+    # Don't source yet - not needed for current functionality
+    # Can be enabled later if we need get_feature_paths()
+    :
+fi
+
+# List of packages tracked for version synchronization
+# NOTE: This is the single source of truth. When adding a new package:
```
Comment (nicholaspsmith) [inline .specify/scripts/bash/update-agent-context.sh]: No. The package.json is the single source of truth. We should parse the "dependencies" array from that file in order to generate the technologies list in CLAUDE.md
